homeassistant/components/shelly/light.py

```diff
@@ -204,6 +217,30 @@ def supported_color_modes(self) -> set | None:
         """Flag supported color modes."""
         return self._supported_color_modes
 
+    @property
+    def effect_list(self):
+        """Lights RGB effects list."""
```

Comment:
```suggestion
    def effect_list(self) -> list[str] | None:
        """Return the list of supported effects.""" 
        if not self.supported_features & SUPPORT_EFFECT:
            return None
```
Follow the definition of light base component: [Base light effect list](https://github.com/home-assistant/core/blob/8e38f26978073f18cc0bc59d9c8b6b917cb5c6e0/homeassistant/components/light/__init__.py#L669) and make sure we don't return list for devices which does not support effects